src/lib/multicall/executeMulticallWorker.ts

```diff
@@ -44,20 +51,25 @@ export async function executeMulticallWorker(chainId: number, request: Multicall
     PRODUCTION_PREVIEW_KEY: localStorage.getItem(PRODUCTION_PREVIEW_KEY),
   });
 
-  const { promise, resolve, reject } = promiseWithResolvers();
+  const { promise, resolve, reject } = promiseWithResolvers<MulticallResult<any> | undefined>();
   promises[id] = { resolve, reject };
 
-  const escapePromise = sleep(2_000).then(() => "timeout");
+  const internalMulticallTimeout = MAX_TIMEOUT;
+  const bufferTimeout = 500;
```

Comment:
is it FREQUENT_UPDATE_INTERVAL - FREQUENT_MULTICALL_REFRESH_INTERVAL
or where it come from?

Reply:
its just a small gap to ensure that worker (that has MAX_TIMEOUT timeout inside) finishes calling public rpc. For example if worker did not start because of being busy by synchronous tasks for a little.